query-translator/src/translate.rs

```diff
@@ -68,6 +73,40 @@ impl ToConstraint for ColumnConstraint {
                 Constraint::equal(qa.to_column(), ColumnOrExpression::Value(tv)),
             EqualsColumn(left, right) =>
                 Constraint::equal(left.to_column(), right.to_column()),
+            EqualsPrimitiveLong(table, value) => {
+                let value_column = QualifiedAlias(table.clone(), DatomsColumn::Value).to_column();
+                let tag_column = QualifiedAlias(table, DatomsColumn::ValueTypeTag).to_column();
+                Constraint::And {
+                    constraints: vec![
+                        Constraint::equal(value_column, ColumnOrExpression::Value(TypedValue::Long(value))),
+                        Constraint::In {
+                            left: tag_column,
+                            list:
+                                if value < 0 {
```

Comment:
The query input is non-symbolic: we get an integer, and now we're trying to find out which types it could match. The best I think we could do here is to push this logic into `ValueType` as a static function: "which types could be represented by this value?".